File: myai/core/llm.py
import requests
from typing import List, Dict, Optional
from myai.config import OLLAMA_URL, DEFAULT_MODEL


def ollama_chat(
    messages: List[Dict[str, str]],
    model: str = DEFAULT_MODEL,
    stream: bool = False,
) -> str:
    url = f"{OLLAMA_URL}/api/chat"
    payload = {
        "model": model,
        "messages": messages,
        "stream": stream,
    }
    r = requests.post(url, json=payload)
    if r.status_code != 200:
        print(f"DEBUG: Ollama Error {r.status_code}: {r.text}")
    r.raise_for_status()

    if stream:
        full = ""
        for line in r.iter_lines():
            if not line:
                continue
            data = line.decode("utf-8")
            full += data
        return full

    data = r.json()
    return data["message"]["content"]


def simple_completion(prompt: str, model: str = DEFAULT_MODEL) -> str:
    return ollama_chat(
        messages=[{"role": "user", "content": prompt}],
        model=model,
        stream=False,
    )

File: src/utils/llm.ts
import { Logger } from "./logger.js";

const logger = new Logger('llm.log');

export async function chatWithOllama(prompt: string, system: string = "", model: string = "qwen2.5-coder:1.5b"): Promise<string> {
    try {
        const response = await fetch("http://127.0.0.1:11434/api/generate", {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({
                model: model,
                prompt: prompt,
                system: system,
                stream: false
            })
        });
        
        if (!response.ok) {
            throw new Error(`Ollama API returned ${response.status}: ${response.statusText}`);
        }
        
        const data = await response.json();
        return data.response || "";
    } catch (e: any) {
        logger.log(`Ollama Error: ${e.message}`, 'error');
        return `[LLM Error: ${e.message}. Is Ollama running?]`;
    }
}

export async function chatWithOllamaHistory(messages: { role: string, content: string }[], model: string = "qwen2.5-coder:1.5b"): Promise<string> {
    try {
        const response = await fetch("http://127.0.0.1:11434/api/chat", {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({
                model: model,
                messages: messages,
                stream: false
            })
        });

        if (!response.ok) {
            throw new Error(`Ollama API returned ${response.status}: ${response.statusText}`);
        }

        const data = await response.json();
        if (!data.message || !data.message.content) {
            logger.log(`Ollama returned empty response: ${JSON.stringify(data)}`, 'warn');
            return "[Empty response from Ollama]";
        }
        return data.message.content;
    } catch (e: any) {
        logger.log(`Ollama Chat Error: ${e.message}`, 'error');
        return `[LLM Error: ${e.message}]`;
    }
}
